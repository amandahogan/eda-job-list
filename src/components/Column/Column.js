import React, { Component } from 'react'
import { Grid } from 'semantic-ui-react'

import './Column.css'
import Company from '../Company'

class Column extends Component {
  render () {
    return (
      <Grid.Column className='Column' width={8}>
        <Company />
      </Grid.Column>
    )
  }
}

export default Column
