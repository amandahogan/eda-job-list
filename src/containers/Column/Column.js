import React, { Component, PropTypes } from 'react'
import { connect } from 'react-redux'
import { Container, Grid } from 'semantic-ui-react'

import { removeColumn } from '../../actions/columns'
import { showDetails } from '../../actions/details'
import Company from '../../components/Company'
import Filter from '../../components/Filter'
import Sort from '../../components/Sort'
import { getCompanies } from '../../selectors/companies'
import Actions from './Actions'
import './Column.css'

class Column extends Component {
  render () {
    const actions = [
      {
        onClick: this.props.closeThisColumn,
        text: 'Close Column'
      }
    ]
    return (
      <Container>
        <Actions actions={actions} />
        <Filter />
        <Grid.Column className='Column' computer={7} mobile={16} tablet={16}>
          <Sort />
          {this.props.companies.map((company, index) => {
            return (
              <Company
                company={company}
                key={index}
                showDetails={() => {
                  this.props.showDetails(company.id)
                }}
              />
            )
          })}
        </Grid.Column>
      </Container>
    )
  }
}

Column.propTypes = {
  column: PropTypes.shape({
    id: PropTypes.number.isRequired
  }).isRequired
}

export default connect(
  state => {
    return {
      companies: getCompanies(state)
    }
  },
  (dispatch, ownProps) => {
    return {
      closeThisColumn: () => {
        dispatch(removeColumn(ownProps.column.id))
      },
      showDetails: companyId => {
        dispatch(showDetails(companyId))
      }
    }
  }
)(Column)
