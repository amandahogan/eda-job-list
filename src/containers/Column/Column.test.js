import { shallow } from 'enzyme'
import React from 'react'
import configureStore from 'redux-mock-store'

import Column from './Column'

const mockStore = configureStore()

it('renders without crashing', () => {
  const initialState = {
    companies: [],
    tags: []
  }
  const store = mockStore(initialState)
  const column = {
    id: 0,
    sortBy: 'Name'
  }
  shallow(
    <Column store={store} column={column} />
  )
})

it('matches snapshot', () => {
  const initialState = {
    companies: [],
    tags: []
  }
  const store = mockStore(initialState)
  const column = {
    id: 0,
    sortBy: 'Name'
  }
  expect(
    shallow(
      <Column store={store} column={column} />
  ).getNode()).toMatchSnapshot()
})
