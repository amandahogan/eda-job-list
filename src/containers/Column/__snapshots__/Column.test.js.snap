exports[`test matches snapshot 1`] = `
<Column
  categories={undefined}
  changeSortBy={[Function]}
  closeThisColumn={[Function]}
  column={
    Object {
      "id": 0,
      "sortBy": "Name",
    }
  }
  companies={Array []}
  showDetails={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": null,
    }
  }
  tags={Array []} />
`;
